src/server.ts

```diff
@@ -16,12 +16,13 @@ export class GCPPubSubStrategy
   extends Server
   implements CustomTransportStrategy
 {
-  protected readonly logger = new Logger(GCPPubSubStrategy.name);
+  protected readonly logger: Logger;
   private readonly subscriptions: Record<string, Subscription> = {};
 
   constructor(private readonly opts: GCPPubSubServerOptions) {
     super();
     this.deserializer = opts.deserializer ?? new JSONDeserializer();
+    this.logger = opts.logger ?? new Logger(GCPPubSubStrategy.name);
```

Comment:
I'm wondering if there are any official recommendations on properly doing this. Could you please point me to how logger overrides are made in official transports, please?

Reply:
sure, i ll look around if there are any recommendations

Reply:
Thanks, I think this should be a pretty common part of all transports. If you won't find any then let's merge it, I just want to be sure

Reply:
No problem. Sure, logger implementation substitution should work just fine here